### PR TITLE
[New Rule] Potential Persistence via Periodic Tasks

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/897

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2021/01/21"
+maturity = "production"
+updated_date = "2021/01/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or modification of the Periodics default configuration. Adversaries may abuse the periodic jobs
+to execute malicious code persistently.
```
Comment (peasead) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:10]:
```suggestion
Identifies the creation or modification of the default configuration for the periodic facility.
```

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Potential Persistence via Periodic Tasks"
```
Comment (peasead) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:16]:
```suggestion
name = "Potential Persistence via Periodic Tasks"
note = "Adversaries may abuse the periodic jobs to execute malicious code or maintain persistence."
```

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+references = [
+    "https://opensource.apple.com/source/crontabs/crontabs-13/private/etc/defaults/periodic.conf.auto.html",
+    "https://www.oreilly.com/library/view/mac-os-x/0596003706/re328.html",
+    "https://github.com/D00MFist/PersistentJXA/blob/master/PeriodicPersist.js",
+]
+risk_score = 47
+rule_id = "48ec9452-e1fd-4513-a376-10a1a26d2c83"
+severity = "medium"
```
Comment (peasead) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:24]:
```suggestion
severity = "low"
```

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Potential Persistence via Periodic Tasks"
+references = [
+    "https://opensource.apple.com/source/crontabs/crontabs-13/private/etc/defaults/periodic.conf.auto.html",
+    "https://www.oreilly.com/library/view/mac-os-x/0596003706/re328.html",
+    "https://github.com/D00MFist/PersistentJXA/blob/master/PeriodicPersist.js",
+]
+risk_score = 47
```
Comment (peasead) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:22]:
```suggestion
risk_score = 21
```

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+rule_id = "48ec9452-e1fd-4513-a376-10a1a26d2c83"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+type = "query"
+
+query = '''
+event.category :"file" and not event.type:"deletion" and
+ file.path :(/private/etc/periodic/* or /private/etc/defaults/periodic.conf or /private/etc/periodic.conf)
```
Comment (peasead) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:30]:
```suggestion
event.category:"file" and not event.type:"deletion" and
 file.path:(/private/etc/periodic/* or /private/etc/defaults/periodic.conf or /private/etc/periodic.conf)
```

**Review by peasead — APPROVED** [review]
LGTM

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 6]
```diff
@@ -0,0 +1,50 @@
+[metadata]
+creation_date = "2021/01/21"
+maturity = "production"
+updated_date = "2021/01/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or modification of the default configuration for the periodic facility.
```
Comment (bm11100) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:9]:
```suggestion
Identifies the creation or modification of the default configuration for periodic tasks. Adversaries may abuse periodic tasks to execute malicious code or maintain persistence.
```

Comment (bm11100) [reply rules/macos/persistence_periodic_tasks_file_mdofiy.toml:9]: re lint

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Potential Persistence via Periodic Tasks"
+note = "Adversaries may abuse the periodic jobs to execute malicious code or maintain persistence."
```
Comment (bm11100) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:16]:
```suggestion

```
We should remove this from the notes field and add it to the description (I made a suggestion).

File: rules/macos/persistence_periodic_tasks_file_mdofiy.toml [hunk 8]
```diff
@@ -0,0 +1,49 @@
+[metadata]
+creation_date = "2021/01/21"
+maturity = "production"
+updated_date = "2021/01/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or modification of the default configuration for periodic tasks. Adversaries may abuse periodic tasks to execute malicious code or maintain persistence.
```
Comment (threat-punter) [inline rules/macos/persistence_periodic_tasks_file_mdofiy.toml:9]: This rule needs to be linted

**Review by threat-punter — APPROVED** [review]
👍

**Pull request merged**